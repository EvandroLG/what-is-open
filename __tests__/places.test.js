import fixture from '../__mocks__/restaurants';
import Places from '../src/places';

describe('places', () => {
    it('should update latitude and longitude with the current position', (next) => {
        const geo = global.navigator.geolocation;
        const mockData = {
            coords: {
                latitude: 53.278453400000004,
                longitude: -6.2164291,
            },
        };

        geo.getCurrentPosition.mockImplementation(fn => {
            fn(mockData);
        });

        const places = new Places();
        places.getCurrentPosition().then(() => {
            const { latitude, longitude } = mockData.coords;
            expect(places.coords.lat).toEqual(latitude);
            expect(places.coords.lng).toEqual(longitude);
            next();
        });
    });

    it('should render properly when api has datas to return', () => {
        document.body.innerHTML = '<div id="js-list"></div>';
        const places = new Places();
        const services = places.placesServices;

        services.nearbySearch.mockImplementation((request, fn) => {
            fn(fixture, 'OK');
        });

        google.maps.places.PlacesServiceStatus.OK = 'OK';

        places.searchByOpenPlaces('restaurant');

        const html = document.getElementById('js-list').innerHTML;

        expect(html).toMatchSnapshot();
    });
});
