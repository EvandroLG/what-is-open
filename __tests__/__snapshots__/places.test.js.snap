// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`places should not render anything when api status was different than ok 1`] = `
"
            <ul>
            
                        <li class=\\"list__item\\">
                            <h2 style=\\"background-image: url(undefined)\\" class=\\"list__name\\">
                                Italian Restaurant
                            </h2>
                            <span class=\\"list__address\\">
                                Stillorgan Road, Dublin
                            </span>
                        </li>
                    
                        <li class=\\"list__item\\">
                            <h2 style=\\"background-image: url(undefined)\\" class=\\"list__name\\">
                                Japonese Restaurant
                            </h2>
                            <span class=\\"list__address\\">
                                Merrion Road, Dublin
                            </span>
                        </li>
                    
                        <li class=\\"list__item\\">
                            <h2 style=\\"background-image: url(undefined)\\" class=\\"list__name\\">
                                Brazilian Restaurant
                            </h2>
                            <span class=\\"list__address\\">
                                Camden Street Lower, Dublin
                            </span>
                        </li>
                    
            </ul>
        "
`;
